library/L1_Peripheral/msp432p401r/system_controller.hpp

```diff
@@ -180,21 +186,24 @@ class SystemController final : public sjsu::SystemController
   {
     SetClockDivider(Clock::kSubsystemMaster, peripheral_divider);
   }
+
   /// @returns The current clock divider used for the subsystem master clock.
   uint32_t GetPeripheralClockDivider(const PeripheralID &) const override
   {
-    constexpr uint8_t kDividerValues[] = { 1, 2, 4, 8, 16, 32, 64, 128 };
-    const uint8_t kDividerSelect       = static_cast<uint8_t>(
+    constexpr uint32_t kDividerValues[] = { 1, 2, 4, 8, 16, 32, 64, 128 };
+    const uint32_t kDividerSelect =
         bit::Extract(clock_system->CTL1,
                      Control1Register::kSubsystemClockDividerSelect.position,
-                     Control1Register::kSubsystemClockDividerSelect.width));
+                     Control1Register::kSubsystemClockDividerSelect.width);
```

Comment:
You should need to use `.position` and `.width` here. Just put in `Control1Register::kSubsystemClockDividerSelect`, and that should work.